### DISCUSSION TITLE
How to get the fresh state of page when hit back button of browser

### DISCUSSION THREAD

**Discussion by Maninder-cell** [inertiajs/inertia#769] [Help]
Hi,

Currently, I'm working on laravel eCommerce application with vuejs and interia. Now the only problem is that when the user clicks on the browser's back button, it shows the previous state of the page, not the new fresh state of the page.

e.g

page A shows the price of each product in the table.
when the admin user clicks on any product on page A redirect to update the product page B and when the user updates the product price and hits back. Lead the user to page A again and it shows the previous price of the product in the table, not the new price. basically, it showing the previous value not the fresh value. How to get a fresh page when the user hits the back button.

**Comment by freakzlike** [comment]
There is an issue which describes your problem. It also contains a workaround snippet, which you can add to your code. See https://github.com/inertiajs/inertia/issues/565

**Comment by 300code** [comment]
that super, popstate for back and forward in browser history  if you need more control to do it as preserving

**Comment by EriksonEnterprises** [comment]
I know this is kinda old but, the only fix I found is history encryption 

https://inertiajs.com/docs/v3/security/history-encryption

Also available on V2

Then you can just call the clearHistory on the page/render that you want to erase prior history (stale data), and it'll reset the history, so when you click back, it'll be fresh

You can make it global in config/inertia.php by using:

`'history' => [
'encrypt' => true,
]`

(At least in V3, hopefully same for V2)
